### ISSUE TITLE
why a kernel like CUDAFunctor_add appears when testing MixedFusedRMSNorm?

### ISSUE THREAD

**Issue by HangJie720** [NVIDIA/apex#1753]
When I execute the following python code, I don’t know why a kernel like CUDAFunctor_add appears.

`import torch
from apex.normalization import MixedFusedRMSNorm
from torch.autograd import Variable
import time

datatype = torch.bfloat16
input = torch.randn([1024, 8192], dtype=datatype).cuda()
input1 = torch.rand_like(input)
input2 = torch.rand_like(input)
input1.requires_grad=True
input2.requires_grad=True

input1_fuse = input1.detach().requires_grad_(True)
input2_fuse = input2.detach().requires_grad_(True)

input1_nofuse = input1.detach().requires_grad_(True)
input2_nofuse = input2.detach().requires_grad_(True)

fwd_thresholds=dict(rtol=1e-7, atol=1e-7)
bwd_thresholds=dict(rtol=1e-7, atol=1e-7)

weight = torch.ones([8192], dtype=datatype, device='cuda:0')
grad = torch.rand_like(input)

for i in range(10):
    norm = MixedFusedRMSNorm(input.size()[1:]).to(device='cuda', dtype=datatype)

    input3 = input1_nofuse
    output_ = norm(input3)
    output_ = output_.to('cuda:0')
    output_.backward(gradient=grad)
`
run with nvprof tool to show kernel:

`root@sw-10-113-1-25:/home/t# nvprof python3 test_rmsnorm.py
==45097== NVPROF is profiling process 45097, command: python3 test_rmsnorm.py
==45097== Profiling application: python3 test_rmsnorm.py
==45097== Profiling result:
            Type  Time(%)      Time     Calls       Avg       Min       Max  Name
 GPU activities:   77.01%  12.695ms        11  1.1541ms  3.3270us  12.661ms  [CUDA memcpy HtoD]
                    8.97%  1.4793ms        10  147.93us  144.61us  151.84us  void cuComputeGradInput<c10::BFloat16, float, c10::BFloat16>(c10::BFloat16 const *, c10::BFloat16 const *, int, int, float const *, float const *, float, c10::BFloat16 const *, c10::BFloat16*, bool)
                    6.42%  1.0576ms        10  105.76us  103.71us  108.00us  void cuApplyRMSNorm<c10::BFloat16, float, c10::BFloat16>(c10::BFloat16*, float*, c10::BFloat16 const *, int, int, float, c10::BFloat16 const *)
                    3.29%  541.82us        10  54.182us  53.183us  55.552us  void cuComputePartGradGammaBeta<c10::BFloat16, float, c10::BFloat16>(c10::BFloat16 const *, c10::BFloat16 const *, int, int, float const *, float const *, float, float*, float*, bool)
                    2.99%  492.06us         9  54.673us  53.376us  55.584us  void at::native::vectorized_elementwise_kernel<int=4, at::native::CUDAFunctor_add<c10::BFloat16>, at::detail::Array<char*, int=3>>(int, c10::BFloat16, at::native::CUDAFunctor_add<c10::BFloat16>)
                    0.94%  154.72us         3  51.573us  51.072us  52.384us  _ZN2at6native55_GLOBAL__N__722798bb_22_DistributionUniform_cu_f2fea07d43distribution_elementwise_grid_stride_kernelIfLi4EZNS0_9templates4cuda21uniform_and_transformIN3c108BFloat16EfLm4EPNS_17CUDAGeneratorImplEZZZNS4_14uniform_kernelIS9_EEvRNS_18TensorIteratorBaseEddT_ENKUlvE_clEvENKUlvE2_clEvEUlfE_EEvSC_T2_T3_EUlP24curandStatePhilox4_32_10E0_ZNS1_27distribution_nullary_kernelIS7_fLi4ES9_SL_SG_EEvSC_SH_RKSI_T4_EUlifE_EEviNS_15PhiloxCudaStateET1_SH_
                    0.36%  59.327us        10  5.9320us  5.5040us  6.6240us  void cuComputeGradGammaBeta<float, c10::BFloat16>(float const *, float const *, int, int, int, c10::BFloat16*, c10::BFloat16*, bool)
                    0.03%  4.2880us         1  4.2880us  4.2880us  4.2880us  void at::native::vectorized_elementwise_kernel<int=4, at::native::FillFunctor<c10::BFloat16>, at::detail::Array<char*, int=1>>(int, c10::BFloat16, at::native::FillFunctor<c10::BFloat16>)
`
